### ISSUE TITLE
Comptime type body emission isn't visible inside generic types

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1383]
As an example, the code below doesn't compile:

```
		struct Test4<T>
		{
			[Comptime, OnCompile(.TypeInit)]
			public static void Generator()
			{
				Compiler.EmitTypeBody(typeof(Self), "int test = 0;");
			}

			public void Test() mut
			{
				test = 1;
			}
		}
```

Tested with https://github.com/beefytech/Beef/commit/8f7f7cba21fe9b1bc27d20d773eca2772bd44f13

**Comment by disarray2077** [comment]
The below code also doesn't compile:
```
 		struct Test8Attribute : Attribute, IComptimeTypeApply
 		{
			[Comptime]
			public void ApplyToType(Type type)
			{
				Compiler.EmitTypeBody(type, "int test = 0;");
			}
		}

		[Test8]
		struct STest8<T>
		{
			public void Test() mut
			{
				test = 0;
			}
		}
```
so this isn't specific to `OnCompile`...

**Comment by disarray2077** [comment]
Actually... I just remembered that recently I wrote `GenerateWrapperAttribute` which does comptime type body emission inside a generic type so the issue isn't this. Looks like the issue is that the emitted variable declaration isn't visible inside the class because the following seems to work:
```
STest8<int> a = default;
a.[Friend]test = 0;
```

**Comment by bfiete** [comment]
Comptime doesn't run on unspecialized generic types (ie: it runs on `Test<int>` but not `Test<T>`). Technically unspecialized code doesn't even emit any instructions because there's no concrete types to assign to instances of `T`... but I think that might be possible to solve.

**Comment by bfiete** [comment]
Fixed at 4e18517c036089d3b6d5175832f7df09bc60f742

There is some caveats to not having a concrete `T` when running `Generate`, but I think overall it does what you'd expect. There's also a `typeof(T).IsGenericParam` to detect that unspecialized case.
